db/db_iterator_test.cc

```diff
@@ -4734,6 +4739,76 @@ TEST_P(DBMultiScanIteratorTest, ReseekAcrossBlocksSameUserKey) {
   }
 }
 
+TEST_P(DBMultiScanIteratorTest, AsyncPrefetchAcrossMultipleFiles) {
```

Comment:
Can we test multiple levels (L0 + some non-L0) as well? And also with DeleteRange and external file ingestion. There are equivalent ones for the non-async MultiScan in this file as well as table_test.cc